migen/genlib/resetsync.py

```diff
@@ -4,8 +4,21 @@
 
 __all__ = ["AsyncResetSynchronizer"]
 
-
 class AsyncResetSynchronizer(Special):
+    """AsyncResetSynchronizer
+    Add a reset signal to the provided ClockDomain synchronized into the desired ClockDomain.
+    Synchronizing is done by adding two D-Flip-Flops in series with the ClockDomain clock.
```

Comment:
What are "flip-flops in series with a clock"?
Anyway the way this really works is using two flip-flops with asynchronous set and synchronous clear, see e.g. https://m.eet.com/media/1121857/chapter2_clocks_resets-03.pdf chapter 2.6.5 "Reset Synchronizer". Your comments do not explain that.